File: programming/java/02-programacao-orientada-objetos/05-interfaces/exemplo/iaula-interface/src/animal/Animal.java
package animal;

public class Animal {
    private String nome;
    private String especie;

    // Construtor
    public Animal(String nome, String especie) {
        this.nome = nome;
        this.especie = especie;
    }

    // Getters
    public String getNome() {
        return nome;
    }
    public String getEspecie() {
        return especie;
    }
    // Setters
    public void setNome(String nome) {
        this.nome = nome;
    }
    public void setEspecie(String especie) {
        this.especie = especie;
    }

    public void emitirSom() {
        System.out.println("O animal está emitindo um som.");
    }
}

File: programacao/php/02-poo/05/Feito em aula/mundocao/03 - Classe_abstract_construct/Classe_abstract_animal/Animal.php
<?php 

    abstract class Animal {
        // Atributos
        private string $especie;
        private int $numeroPatas;

        protected function setEspecie(string $especie)
        {
            $this->especie = $especie;
        }

        protected function getEspecie() : string
        {
            return $this->especie;
        }

        protected function setNumeroPatas(int $numeroPatas)
        {
            $this->numeroPatas = $numeroPatas;
        }

        protected function getNumeroPatas() : int
        {
            return $this->numeroPatas;
        }
    }

?>